### PR TITLE
Add Elements (+spectral element)

### PR DESCRIPTION
**Pull request by int-ptr-ptr**

### CODE REVIEW

File: fastfem/elements/spectral_element.py [hunk 1]
```diff
@@ -0,0 +1,666 @@
...
+                local[0] ** np.arange(Np1),
+                local[1] ** np.arange(Np1),
+            )
+
+        e = l2g(local) - target
+        F = 0.5 * np.sum(e**2)
+
+        # linsearch on gradient descent
+        def linsearch(r, step):
+            ARMIJO = 0.25
```
[the remark this answers is not in the document]
Comment (int-ptr-ptr) [reply fastfem/elements/spectral_element.py:330]: I am deprecating this.

**Comment by int-ptr-ptr** [comment]
There are a few tests that could be added, but for now, this has all the baseline features with full test coverage. We need to be able to incorporate this with the mesher, and this should be all set up to do that.

**Comment by int-ptr-ptr** [comment]
If you were in the middle of reviewing, sorry.

I've sped up the field tests and added linear triangular elements.

**Review by sacha-escudier — APPROVED** [review]
Everything is very well done, thank you! I have nothing much to say right now, I will ask you more questions as I initiate the plotter

**Review by sinaatalay — APPROVED** [review]
Thank you!

**Pull request merged**